### ISSUE TITLE
CTB relation editor relations are  not accessible for screen readers or playwright tests

### ISSUE THREAD

**Issue by innerdvations** [strapi/strapi#21420]
### Required System information

Environment: development
OS: darwin-arm64
Strapi Version: 5.0.0
Node/Yarn Version: yarn/4.3.1 npm/? node/v20.9.0 darwin arm64
Edition: Enterprise
Database: sqlite

### Describe the bug

The relations editor is not very accessible for screen readers, because the only thing denoting a selected item is a stylistic color change. The selected item should have an aria attribute (aria-pressed?) to make it clear which one is selected.

I noticed because I was writing an e2e test for this, and there was no way to check which button was selected. So it would assist in writing automated tests as well.

### Steps to reproduce the behavior

View the html in inspector while clicking the relations in the CTB

### Expected behavior

I expect an aria attribute or a radio select group to be used

### Screenshots

![image](https://github.com/user-attachments/assets/aaca275e-d310-4904-9408-d3506b5ed088)

### Code snippets

If applicable, add code samples to help explain your problem.
